include/NeoFOAM/core/database/fieldCollection.hpp

```diff
@@ -0,0 +1,383 @@
+ * @brief Validates a FieldDocument.
+ *
+ * This function validates a FieldDocument by checking if it contains the required fields.
+ *
+ * @param doc The Document to validate.
+ * @return true if the Document is valid, false otherwise.
+ */
+bool validateFieldDoc(const Document& doc);
+
+/**
+ * @class FieldDocument
+ * @brief A class representing a field document in a database.
+ *
+ * The FieldDocument class represents a field document in a database. It is a subclass of the
+ * Document class and provides additional functionality for accessing field-specific data.
+ */
+class FieldDocument
+{
+public:
+
+    /**
+     * @brief Constructs a FieldDocument with the given field and metadata.
+     *
+     * @tparam geoField The type of the field.
+     * @param field The field to store in the document.
+     * @param timeIndex The time index of the field.
+     * @param iterationIndex The iteration index of the field.
+     * @param subCycleIndex The sub-cycle index of the field.
+     */
+    template<class geoField>
+    FieldDocument(
+        const geoField& field,
+        std::int64_t timeIndex,
+        std::int64_t iterationIndex,
+        std::int64_t subCycleIndex
+    )
+        : doc_(
+              Document(
+                  {{"name", field.name},
+                   {"timeIndex", timeIndex},
+                   {"iterationIndex", iterationIndex},
+                   {"subCycleIndex", subCycleIndex},
+                   {"field", field}}
+              ),
+              validateFieldDoc
+          )
+    {}
+
+    /**
+     * @brief Constructs a FieldDocument with the given Document.
+     *
+     * @param doc The Document to construct the FieldDocument from.
+     */
+    FieldDocument(const Document& doc);
+
+    /**
+     * @brief Retrieves the underlying Document.
+     *
+     * @return Document& A reference to the underlying Document.
+     */
+    Document& doc();
+
+    /**
+     * @brief Retrieves the underlying Document (const version).
+     *
+     * @return const Document& A const reference to the underlying Document.
+     */
+    const Document& doc() const;
+
+
+    /**
+     * @brief Retrieves the unique identifier of the field collection.
+     *
+     * @return A string representing the unique identifier.
+     */
+    std::string id() const;
+
+    /**
+     * @brief Retrieves the type name of the field.
+     *
+     * @return A string representing the type name.
+     */
+    static std::string typeName();
+
+    /**
+     * @brief Retrieves the field from the document.
+     *
+     * @tparam geoField The type of the field.
+     * @return A reference to the field.
+     */
+    template<class geoField>
+    geoField& field()
+    {
+        return doc_.get<geoField&>("field");
+    }
+
+    /**
+     * @brief Retrieves the field from the document (const version).
+     *
+     * @tparam geoField The type of the field.
+     * @return A const reference to the field.
+     */
+    template<class geoField>
```

Comment:
Typenames start with capital letter.
```suggestion
    template<class FieldType>
```

Reply:
This needs to replaced for all occurrences or the clang-tidy will complain.